Operator + Forwarder docs (#1101)
smartcontractkit/documentation: src/pages/chainlink-nodes/contracts/receiver.md

---
layout: ../../../layouts/MainLayout.astro
section: nodeOperator
date: Last Modified
title: "Receiver"
---

[AuthorizedReceiver](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/AuthorizedReceiver.sol) is an abstract contract inherited by [Operator](/chainlink-nodes/contracts/operator) and [Forwarder](/chainlink-nodes/contracts/forwarder) contracts.

## Api Reference

### Methods

#### setAuthorizedSenders

```solidity
function setAuthorizedSenders(address[] senders) external
```

> Sets the fulfillment permission for a given node. Use `true` to allow, `false` to disallow.

Reviewer: This section is not very clear to me what it means. Can you please clarify?

Author: based on https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/AuthorizedReceiver.sol#L13

---
layout: ../../../layouts/MainLayout.astro
section: nodeOperator
date: Last Modified
title: "Receiver"
---

> [AuthorizedReceiver](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/AuthorizedReceiver.sol) is an abstract contract inherited by [Operator](/chainlink-nodes/contracts/operator) and [Forwarder](/chainlink-nodes/contracts/forwarder) contracts.

Reviewer 2: ```suggestion
[AuthorizedReceiver](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/AuthorizedReceiver.sol) is an abstract contract inherited by [operator](/chainlink-nodes/contracts/operator) and [forwarder](/chainlink-nodes/contracts/forwarder) contracts.
```

:::note

> Calling [setAuthorizedSenders](#setauthorizedsenders) has a different effect depending if it is called from an [Operator](/chainlink-nodes/contracts/operator) or a [Forwarder](/chainlink-nodes/contracts/forwarder) contract:

Reviewer 2: ```suggestion
Calling [setAuthorizedSenders](#setauthorizedsenders) has a different effect depending if it is called from an [operator](/chainlink-nodes/contracts/operator) or a [forwarder](/chainlink-nodes/contracts/forwarder) contract:
```

> - Forwarder contracts' owners allow authorized senders to call [forward](/chainlink-nodes/contracts/forwarder#forward).

Reviewer 2: ```suggestion
- Owners of forwarder contracts allow authorized senders to call [forward](/chainlink-nodes/contracts/forwarder#forward).
```

> - Operator contracts' owners allow authorized senders to call [fulfillOracleRequest](/chainlink-nodes/contracts/operator#fulfilloraclerequest) and [fulfillOracleRequest2](/chainlink-nodes/contracts/operator#fulfilloraclerequest2)

Reviewer 2: ```suggestion
- Owners of operator contracts allow authorized senders to call the [fulfillOracleRequest](/chainlink-nodes/contracts/operator#fulfilloraclerequest) and [fulfillOracleRequest2](/chainlink-nodes/contracts/operator#fulfilloraclerequest2) methods.
```

Avoiding possessive nouns is usually easier to parse mentally.

  :::

## Api Reference

### Methods

#### setAuthorizedSenders

```solidity
function setAuthorizedSenders(address[] senders) external
```

Sets the fulfillment permission for a given node. Use `true` to allow, `false` to disallow.

> Emit [AuthorizedSendersChanged](#authorizedsenderschanged) event.

Reviewer 2: ```suggestion
Emits an [AuthorizedSendersChanged](#authorizedsenderschanged) event.
```

##### Parameters

| Name    | Type      | Description                                    |
| ------- | --------- | ---------------------------------------------- |
| senders | address[] | The addresses of the authorized Chainlink node |

#### getAuthorizedSenders

```solidity
function getAuthorizedSenders() external view returns (address[])
```

Retrieve a list of authorized senders.

##### Return Values

| Name | Type      | Description        |
| ---- | --------- | ------------------ |
|      | address[] | array of addresses |

#### isAuthorizedSender

```solidity
function isAuthorizedSender(address sender) public view returns (bool)
```

> Use this to check if a node is authorized for fulfilling requests.

Reviewer 2: ```suggestion
Use this to check if a node is authorized to fulfill requests.
```
